.clang-format

```diff
@@ -0,0 +1,169 @@
+---
```

Comment:
Can you avoid dumping the whole clang-format file and use `BasedOnStyle: Google`?

Also, Ubuntu 18.04 comes with clang-format 6, so we need to support this version at least.